i18n/README.md

```diff
@@ -5,11 +5,275 @@
 ## Summary
 
 - [Intentions](#intentions)
+- [TranslatorBuilder](#translatorbuilder)
+  - [Representation](#representation)
+  - [Adding translations](#adding-translations)
+  - [DuplicatedKeyResolution](#duplicatedkeyresolution)
+  - [Settings](#settings)
+  - [Building](#building)
+- [Translator](#translator)
+  - [Basic usage](#basic-usage)
+  - [Variable interpolation](#variable-interpolation)
+  - [Options](#options)
+- [Configuration](#configuration)
+  - [`interpolationDelimiter`](#interpolationdelimiter)
+  - [`pluralMapper`](#pluralmapper)
+  - [`formats`](#formats)
+  - [`missingKeyPolicy`](#missingkeypolicy)
 - [Changelog](#changelog)
 
 ## Intentions
 
-This module provides a simple way to manage internationalization in a Java/Kotlin project.
+This module provides a simple way to manage internationalization in a Java/Kotlin project, with features such as
+variable interpolation and formatting, and automatic loading of translation files
+
+## TranslatorBuilder
+
+Before anyone can use the `Translator` interface, which will be covered in the next section, one has to first build
+an instance of it.
+
+Thankfully, `TranslatorBuilder` exists for this purpose. The goal is to provide it with different sources of translation
+associations. 
+
+### Representation
+
+Indeed, a translation is a key-value pair, where the key is a `TranslationKey` (which is a type alias for
+`String`), and the value is a `TranslationSourceData` (which is a type alias for `Any`) and corresponds to the
+translation itself. This key-value pair is itself associated to a `Locale`.
+
+In fact, a `TranslationKey` is a `String` that must follow a specific format, and here are some examples of valid keys:
+Here are some examples of valid `TranslationKeys`:
+- `my-root.my-node.my-key`
+- `my_root.my_node.my_key`
+- `myRoot.myNode.myKey`
+- `my-root.my-node.my-table.1`
+
```

Comment:
Missing the explanation of the TranslatorSourceData (String, Int, etc., you can reuse the typealias doc)